File: scripts/submit_test.py
#!/usr/bin/env python3
"""示例：通过 Python 脚本提交压测任务"""
import json
import urllib.request

MASTER = "http://localhost:8080"

plan = {
    "name": "demo-ping-test",
    "mode": "STANDALONE",
    "load": {"concurrency": 20, "durationSeconds": 10},
    "requests": [
        {"method": "GET", "url": f"{MASTER}/api/demo/ping"}
    ]
}

req = urllib.request.Request(
    f"{MASTER}/api/tasks",
    data=json.dumps(plan).encode(),
    headers={"Content-Type": "application/json"},
    method="POST",
)
with urllib.request.urlopen(req) as resp:
    result = json.loads(resp.read())
    print("Task submitted:", result)
    task_id = result["data"]["taskId"]
    print(f"Open dashboard: {MASTER}/index.html and enter task ID: {task_id}")

File: scripts/submit_test.js
/** 示例：通过 Node.js 脚本提交压测任务 */
const MASTER = process.env.MASTER_URL || 'http://localhost:8080';

const plan = {
  name: 'demo-ping-test',
  mode: 'DISTRIBUTED',
  load: { concurrency: 50, durationSeconds: 15 },
  requests: [
    { method: 'GET', url: `${MASTER}/api/demo/ping` }
  ]
};

fetch(`${MASTER}/api/tasks`, {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  body: JSON.stringify(plan)
})
  .then(r => r.json())
  .then(result => {
    console.log('Task submitted:', result);
    const taskId = result.data.taskId;
    console.log(`Dashboard: ${MASTER}/index.html  Task ID: ${taskId}`);
  })
  .catch(console.error);
